src/AgentVisibilityContext.tsx

```diff
@@ -0,0 +1,221 @@
+import React, { createContext } from 'react';
+import { isEqual } from 'underscore';
+import {
+  DisplayAction,
+  DisplayStateEntry,
+  HiddenOrHighlightedState,
+  SubAgentDisplayMaps,
+  TopLevelDisplayStatus,
+  UIDisplayEntry,
+  VisibilitySelectionMap,
+  ViewerVisibilityStates,
+} from './constants';
+import {
+  getPayloadForHideAll,
+  getPayloadSubAgentDisplayAction,
+  getPayloadTopLevelDisplayAction,
+  getSelectionStateInfoPayload,
+  getSubAgentDisplayStatePayload,
+  getTopLevelDisplayStatePayload,
+} from './selectors';
+import { SelectionStateInfo, UIDisplayData } from '@aics/simularium-viewer';
+
+interface VisibilityContextType {
+  uiDisplayData: UIDisplayData;
+  currentVisibilityStates: ViewerVisibilityStates;
+  selectionStateInfo: SelectionStateInfo;
+  allAgentsHiddenState: HiddenOrHighlightedState;
+  receiveUIDisplayData: (data: UIDisplayData) => void;
+  updateVisibilityAndSelection: (
+    newValue: VisibilitySelectionMap,
+    hideOrHighlight: DisplayAction
+  ) => void;
+  hideOrRevealAllAgents: (hiddenState: HiddenOrHighlightedState) => void;
+  getTopLevelDisplayStatus: (agent: UIDisplayEntry) => TopLevelDisplayStatus;
+  getSubAgentDisplayMaps: (agent: UIDisplayEntry) => SubAgentDisplayMaps;
+  handleTopLevelDisplayAction: (
+    agent: UIDisplayEntry,
+    hideOrHighlight: DisplayAction
+  ) => void;
+  handleSubAgentDisplayAction: (
+    agent: UIDisplayEntry,
+    hideOrHighlight: DisplayAction,
+    subAgent?: DisplayStateEntry
+  ) => void;
+}
+const { Hide, Highlight } = DisplayAction;
+const { Inactive, Active, Indeterminate } = HiddenOrHighlightedState;
+
+export const VisibilityContext = createContext<VisibilityContextType>({
+  uiDisplayData: [],
+  currentVisibilityStates: { hidden: {}, highlight: {} },
+  selectionStateInfo: {
+    hiddenAgents: [],
+    highlightedAgents: [],
+    colorChange: null,
+  },
+  allAgentsHiddenState: Inactive,
+  receiveUIDisplayData: () => {},
+  updateVisibilityAndSelection: () => {},
+  hideOrRevealAllAgents: () => {},
+  getTopLevelDisplayStatus: () => {
+    return { hidden: Inactive, highlight: Inactive };
+  },
+  getSubAgentDisplayMaps: () => {
+    return { hidden: {}, highlight: {} };
+  },
+  handleTopLevelDisplayAction: () => {},
+  handleSubAgentDisplayAction: () => {},
+});
```

Comment:
These default fallbacks should hopefully never be used as long as we always have a context provider in the tree above the component trying to access the context.